### PR TITLE
Cache version per influxdb instance and reduce ping() calls for every…

### LINKED ISSUE #470: Please remove ping checks from query method
In `InfluxDBImpl::query()` there are two version checks at the top, both pinging the server _*twice*_ (!!!) for the version number:

    public void query(final Query query, final int chunkSize, final Consumer<QueryResult> consumer) {
        if (version().startsWith("0.") || version().startsWith("1.0")) {
            throw new UnsupportedOperationException("chunking not supported");
        }
        ...

Please either remove those version checks completely (I think nobody uses these old versions anymore), or cache the result of `version()`, or query each server only _once_ for it's version number.

### PR DESCRIPTION
**Pull request by majst01**
… query call, closes #470

### CODE REVIEW

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 1]
```diff
@@ -313,7 +314,10 @@ public Pong ping() {
 
   @Override
   public String version() {
-    return ping().getVersion();
+    if (version == "") {
```
[the remark this answers is not in the document]
Comment (majst01) [reply src/main/java/org/influxdb/impl/InfluxDBImpl.java]: You are right, to much different languages, java is getting out of my head by the time :-). Fixed

**Comment by lxhoan** [comment]
I would like to ask how the client can know if the server side is upgraded to a new version ?

**Comment by majst01** [comment]
The client cant, we promise compatibility to 1.1 onwards. We could probably add a check in the constructor which crashes once, but the current code does this check on every query which is awkward. 

What do you think ?

**Comment by lxhoan** [comment]
A check (for server version) in the constructor is also the approach I would like to use for MessagePack so of course I would vote for it. 

For the cached version, it may go a bit confused and inconsistent (when the server side version changes, yet not very likely) for user so I just think we can document the method version() explicitly on this

**Pull request merged**